Modules/_ctypes/callproc.c

```diff
@@ -57,13 +57,27 @@
 #include "Python.h"
 #include "structmember.h"         // PyMemberDef
 
+#include <stdbool.h>
+
 #ifdef MS_WIN32
 #include <windows.h>
 #include <tchar.h>
 #else
 #include "ctypes_dlfcn.h"
 #endif
 
+#ifdef __APPLE__
+/*
+ * The API to query if a shared library is in the shared cache is 
+ * private for now, this should change in beta 4.
+ *
+ * TODO:
+ * - Switch to that API
```

Comment:
this api has been made public.   see updates to https://github.com/python/cpython/pull/21241

Reply:
That's been changed in the current version of the PR.